### PR TITLE
Added errors guide docs for cloud services

### LINKED ISSUE #113: New error codes from Cloud Services plugin
we should add documentation for two error codes:
* `cloudservices-no-url` - this error occurs when there is no cloudservice url in config and no url is passed to cloudservice plugin.
* `cloudservices-no-token` - the same as above but for auth token.

### PR DESCRIPTION
**Pull request by jacekbogdanski**
Added errors guide docs for cloud services.

Closes #113

### CODE REVIEW

File: docs/guide/dev/deep_dive/errors/README.md [hunk 1]
```diff
@@ -23,6 +23,16 @@ This article contains the list of CKEditor error codes and their explanation. Th
 * Description: Incorrect {@linkapi CKEDITOR.config#autoEmbed_widget } value. No widget definition found.
 * Additional data: None.
 
+## cloudservices-no-url
+* Location: `plugins/cloudeservices/plugin.js`
+* Description: Upload url used by {@linkapi CKEDITOR.config.cloudServices} is not set. An url can be provided by configuration {@linkapi CKEDITOR.config#url} property or passed to [Cloud Services](https://ckeditor.com/ckeditor-cloud-services/) plugin {@linkapi CKEDITOR.plugins.cloudeServices.cloudServicesLoader#upload} method.
```
Comment (mlewand) [inline docs/guide/dev/deep_dive/errors/README.md]: Link to the addon will be `https://ckeditor.com/cke4/addon/cloudservices`. You're talking about the plugin here, so you need to link the plugin not the whole CS product itself.

File: docs/guide/dev/deep_dive/errors/README.md [hunk 2]
```diff
@@ -23,6 +23,16 @@ This article contains the list of CKEditor error codes and their explanation. Th
 * Description: Incorrect {@linkapi CKEDITOR.config#autoEmbed_widget } value. No widget definition found.
 * Additional data: None.
 
+## cloudservices-no-url
```
Comment (mlewand) [inline docs/guide/dev/deep_dive/errors/README.md]: There  should be a space between header and unordered list.

File: docs/guide/dev/deep_dive/errors/README.md [hunk 3]
```diff
@@ -23,6 +23,16 @@ This article contains the list of CKEditor error codes and their explanation. Th
...
+* Location: `plugins/cloudeservices/plugin.js`
```
Comment (mlewand) [inline docs/guide/dev/deep_dive/errors/README.md]: There is a typo in path. Check it in your FS it's incorrect.

File: docs/guide/dev/deep_dive/errors/README.md (hunk 1 above)
Comment (mlewand) [inline docs/guide/dev/deep_dive/errors/README.md]: URL is an abbreviation. So there's URL, HTML, CSS etc.

File: docs/guide/dev/deep_dive/errors/README.md [hunk 4]
```diff
@@ -23,6 +23,16 @@ This article contains the list of CKEditor error codes and their explanation. Th
...
+* Description: Upload url used by {@linkapi CKEDITOR.config.cloudServices} is not set. An url can be provided by configuration {@linkapi CKEDITOR.config#url} property or passed to [Cloud Services](https://ckeditor.com/ckeditor-cloud-services/) plugin {@linkapi CKEDITOR.plugins.cloudeServices.cloudServicesLoader#upload} method.
+* Additional data: None.
+
+## cloudservices-no-token
+* Location: `plugins/cloudeservices/plugin.js`
+* Description: Auth token used by {@linkapi CKEDITOR.config.cloudServices} is not set. An token can be provided by configuration {@linkapi CKEDITOR.config#cloudServices_token} property or passed to [Cloud Services](https://ckeditor.com/ckeditor-cloud-services/) plugin {@linkapi CKEDITOR.plugins.cloudeServices.cloudServicesLoader#constructor}.
```
Comment (mlewand) [inline docs/guide/dev/deep_dive/errors/README.md]: "Auth token" -> "Authentication token". This is a public documentation, thus we use full forms.

File: docs/guide/dev/deep_dive/errors/README.md [hunk 5]
```diff
@@ -23,6 +23,18 @@ This article contains the list of CKEditor error codes and their explanation. Th
 * Description: Incorrect {@linkapi CKEDITOR.config#autoEmbed_widget } value. No widget definition found.
 * Additional data: None.
 
+## cloudservices-no-url
+
+* Location: `plugins/cloudservices/plugin.js`
+* Description: Upload URL used by {@linkapi CKEDITOR.plugins.cloudServices.cloudServicesLoader} is not set. An URL can be provided by configuration {@linkapi CKEDITOR.config#url} property or passed to [Cloud Services](https://ckeditor.com/cke4/addon/cloudservices) plugin {@linkapi CKEDITOR.plugins.cloudServices.cloudServicesLoader#upload} method.
```
Comment (mlewand) [inline docs/guide/dev/deep_dive/errors/README.md]: Sentence:

> Upload URL used by CKEDITOR.plugins.cloudServices.cloudServicesLoader is not set.

Is incorrect. Please read the whole change proposed in PR to make sure there are no more issues like that. You want to generally read (sometimes even aloud) things you have written, before pushing your commit.

Comment (mlewand) [reply docs/guide/dev/deep_dive/errors/README.md]: You're right with pointing out that the URL could be customized in `cloudServicesLoader#upload` method, but I feel we're diving a bit too deep here. Let's stick to a more generic relation - I mean, you should talk about Cloud Services plugin and not particular API function names.

So you want to remove part about possible URL customization (this should be documented in API docs of a proper method/argument) and you don't want to relate to a particular API function.

File: docs/guide/dev/deep_dive/errors/README.md [hunk 6]
```diff
@@ -23,6 +23,18 @@ This article contains the list of CKEditor error codes and their explanation. Th
...
+* Additional data: None.
+
+## cloudservices-no-token
+
+* Location: `plugins/cloudservices/plugin.js`
+* Description: Authentication token used by {@linkapi CKEDITOR.plugins.cloudServices.cloudServicesLoader} is not set. An token can be provided by configuration {@linkapi CKEDITOR.config#cloudServices_token} property or passed to [Cloud Services](https://ckeditor.com/cke4/addon/cloudservices) plugin {@linkapi CKEDITOR.plugins.cloudServices.cloudServicesLoader#constructor}.
```
Comment (mlewand) [inline docs/guide/dev/deep_dive/errors/README.md]: "An token" -> "The token"

Comment (mlewand) [reply docs/guide/dev/deep_dive/errors/README.md]: Same thing that it should be more generic.

Comment (mlewand) [reply docs/guide/dev/deep_dive/errors/README.md]: There's a small issue that we don't have yet the docs on setting Cloud Services token. So we don't ave anything to point to ATM, we'll add it later.

**Comment by jacekbogdanski** [comment]
When we finish documentation for Cloud Services we should complete proposed descriptions with correct references like `Refer to the {@link link title} article for more information.`

**Pull request merged**